### PR TITLE
fix istioctl create incorrect namespace for istio control plane.

### LINKED ISSUE #29523: istioctl doesn't create the correct namespace for istio control plane
**Bug description**

currently istioctl will create the namespace of istio operator CR, but the istio control plane will may actually created in different namespace, for example:

```
# cat << EOF > iop.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: ns1
  name: example-istiocontrolplane
spec:
  profile: default
  namespace: istio-system
EOF
# out/linux_amd64/istioctl install -f iop.yaml
This will install the Istio default profile with ["Istio core" "Istiod" "Ingress gateways"] components into the cluster. Proceed? (y/N) y
Detected that your cluster does not support third party JWT authentication. Falling back to less secure first party JWT. See https://istio.io/v1.8/docs/ops/best-practices/security/#configure-third-party-service-account-tokens for details.
  Processing resources for Istio core.                                                                                                                                                                                                      2020-12-09T14:22:44.596260Z	error	installer	failed to create "ServiceAccount/istio-system/istio-reader-service-account": namespaces "istio-system" not found
2020-12-09T14:22:44.676108Z	error	installer	failed to create "ServiceAccount/istio-system/istiod-service-account": namespaces "istio-system" not found
- Processing resources for Istio core.                                                                                                                                                                                                      2020-12-09T14:22:44.981787Z	error	installer	failed to create "Role/istio-system/istiod-istio-system": namespaces "istio-system" not found
...
```

[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

**How was Istio installed?**

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

### PR DESCRIPTION
**Pull request by morvencao**
resolve: https://github.com/istio/istio/issues/29523

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: operator/cmd/mesh/install.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ func InstallManifests(setOverlay []string, inFilenames []string, force bool, dry
 		return nil, err
 	}
 
-	if err := createNamespace(clientset, iop.Namespace, networkName(iop)); err != nil {
+	if err := createNamespace(clientset, v1alpha12.Namespace(iop.Spec), networkName(iop)); err != nil {
```
Comment (shamsher31) [inline operator/cmd/mesh/install.go:207]: will this also handle `values.global.istioNamespace` ?

Comment (morvencao) [reply operator/cmd/mesh/install.go:207]: Yes, see: https://github.com/istio/istio/blob/5ae68fb2df850a13bebba2533b25279c956dd511/operator/pkg/apis/istio/v1alpha1/common.go#L29-L46

**Comment by shamsher31** [comment]
I have used this IOP.
```
$ cat << EOF > iop.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: ns1
  name: example-istiocontrolplane
spec:
  profile: default
EOF
```

Installation completed but throws an error.
```
$ ./out/linux_amd64/istioctl install -f iop.yaml --set hub=gcr.io/istio-testing -d manifests/
This will install the Istio 1.9.0 default profile with ["Istio core" "Istiod" "Ingress gateways"] components into the cluster. Proceed? (y/N) y
✔ Istio core installed                                                                                                                   
✔ Istiod installed                                                                                                                       
✔ Ingress gateways installed                                                                                                             
✔ Installation complete                                                                                                                  
Error: failed to install manifests: failed to create "IstioOperator/ns1/installed-state-example-istiocontrolplane": namespaces "ns1" not found
```

Also, this should install Istio in `ns1` namespace or `istio-system` namespace?

**Comment by morvencao** [comment]
@shamsher31 So this is the problem, should we use the istiooperator CR's namespace as the namespace of istio control plane.
I prefer to create namespace for istio control plane, or we should create both namespaces?

Because in this case, istio control plane will create in `istio-system` namespace, right?

**Comment by shamsher31** [comment]
That's really confusing because we are defining two different namespaces one in `metadat.namespace` and one in `spec.namespace`, which one should we prefer?

> My assumption was if I specify metadata.namespace, IstioOperator should automatically create a namespace for installation like it does for istio-system namespace. https://github.com/istio/istio/issues/29523#issuecomment-741810687

**Pull request closed without merging**